aspnetcore/fundamentals/url-rewriting.md

```diff
@@ -0,0 +1,303 @@
+
+#### URL Redirect to a secure endpoint (permanent)
+Use `AddRedirectToHttpsPermanent()` to redirect insecure requests to the same host and path with secure HTTPS protocol (**https://** on port 465). The middleware will set the status code to 301 (Moved Permanently).
+```csharp
+var options = new RewriteOptions()
+    .AddRedirectToHttpsPermanent();
+
+app.UseRewriter(options);
+```
+The sample application is capable of demonstrating how to use `AddRedirectToHttpsPermanent()`. Add the extension method to the `RewriteOptions()`. Make an insecure request to the application at any URL. In order to see the response in a browser, you will probably need to dismiss a browser security warning that the self-signed certificate is untrusted.
+
+Original Request: **http://localhost:5000/secure**
+
+![Browser window with Developer Tools tracking the requests and responses](url-rewriting/_static/add_redirect_to_https_permanent.png)
+
+#### URL Redirect to a secure endpoint
+Use `AddRedirectToHttps()` to redirect insecure requests to the same host and path with secure HTTPS protocol (**https://**) with the flexibility to choose the status code and port. If the status code is not supplied, the middleware will default to 302 (Found). If the port is not supplied, the middleware will default to `null`, which means the protocol will change to **https://** and the client will access the resource on port 465. The example shows how to set the status code to 301 (Moved Permanently) and change the port to 5001.
+```csharp
+var options = new RewriteOptions()
+    .AddRedirectToHttps(301, 5001);
+
+app.UseRewriter(options);
+```
+The sample application is capable of demonstrating how to use `AddRedirectToHttps()`. Add the extension method to the `RewriteOptions()` with a 301 response status code on port 5001. Make an insecure request to the application at any URL. In order to see the response in a browser, you will probably need to dismiss a browser security warning that the self-signed certificate is untrusted.
+
+Original Request: **http://localhost:5000/secure**
+
+![Browser window with Developer Tools tracking the requests and responses](url-rewriting/_static/add_redirect_to_https.png)
+
+#### URL Rewrite
+Use `AddRewrite()` to create a rules for rewriting URLs. The first parameter will contain your regex for matching on the incoming URL path. The second parameter is the replacement string. The third parameter, `skipRemainingRules: {true|false}`, will indicate to the middleware whether or not to skip additional rewrite rules if the current rule is applied.
```

Comment:
Not that I can see ...

https://github.com/aspnet/BasicMiddleware/blob/dev/src/Microsoft.AspNetCore.Rewrite/Internal/RewriteRule.cs#L37-L46